File: javascript/pe009.js
"use strict"

const stop = 1000;

for (let a = 1; a < stop; a++) {
  for (let b = a + 1; b < stop; b++) {
    for (let c = b + 1; c < stop; c++) {
      if ((a + b +c) === stop) {
        if ((a * a) + (b * b) === (c * c)) {
          console.log(a*b*c);
        }
      }
    }
  }
}

File: perl/pe009.pl
#!/usr/bin/perl -w
use strict;

my $stop = 1000;
for (my $a = 1; $a < $stop; $a++) {
  for (my $b = $a + 1; $b < $stop; $b++) {
    for (my $c = $b + 1; $c < $stop; $c++) {
      if ((($a + $b +$c) == $stop) and (($a * $a) + ($b * $b) == ($c * $c))) {
        print $a*$b*$c,"\n";
        exit(0);
      }
    }
  }
}
exit(1);
